src/ray/common/task/task_spec.h

```diff
@@ -29,11 +29,41 @@
 extern "C" {
 #include "ray/thirdparty/sha256.h"
 }
-
 namespace ray {
-typedef ResourceSet SchedulingClassDescriptor;
+/* typedef ResourceSet SchedulingClassDescriptor; */
 typedef int SchedulingClass;
 
+struct SchedulingClassDescriptor {
+ public:
+  explicit SchedulingClassDescriptor(ResourceSet &rs, FunctionDescriptor &fd, int64_t d)
+      : resource_set(rs), function_descriptor(fd), depth(d) {}
+  ResourceSet resource_set;
+  FunctionDescriptor function_descriptor;
+  int64_t depth;
+
+  bool operator==(const SchedulingClassDescriptor &other) const {
```

Comment:
Use

```
return depth == other.depth && 
    resource_set == other.resource_set && 
    function_descriptor == other.function_descriptor && 
```
for the following reasons:
1. Short-circuit return
2. depth is cheaper to compare